modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -362,6 +367,44 @@ public final boolean isOpaque() {
         return false;
     }
 
+    // we need to know the JavaFX screen of the current AWT graphcisConfiguration
+    private Screen findScreen(GraphicsConfiguration graphicsConfiguration) {
+        Rectangle awtBounds = graphicsConfiguration.getBounds();
+        AffineTransform awtScales = graphicsConfiguration.getDefaultTransform();
+        for (Screen screen : Screen.getScreens()) {
+            if ((Math.abs(screen.getPlatformX() - awtBounds.getX()) < 2.) &&
+                    (Math.abs(screen.getPlatformY() - awtBounds.getY()) < 2.) &&
+                    (Math.abs(screen.getPlatformWidth() - awtScales.getScaleX() * awtBounds.getWidth()) < 2.) &&
+                    (Math.abs(screen.getPlatformHeight() - awtScales.getScaleY() * awtBounds.getHeight()) < 2.)) {
+                return screen;
+            }
+        }
+        return null;
+    }
+
+    private Dimension2D convertSwingToFxPixel(GraphicsConfiguration g, float wx, float wy) {
+        float newx, newy;
+        Screen screen = findScreen(getGraphicsConfiguration());
```

Comment:
Did you mean to use the passed in graphics config, `g`? If not, you might remove the `g` parameter, since it is otherwise unused.